File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> answer = new ArrayList<>();

        for (int i = 0; i < numRows; i++) {
            List<Integer> row = new ArrayList<Integer>();

            row.add(1);  

            if (i > 0) {
                List<Integer> prevRow = answer.get(i - 1);
                for (int j = 1; j < prevRow.size(); j++) {
                    row.add(prevRow.get(j - 1) + prevRow.get(j));
                }
                row.add(1);
            }

            answer.add(row);
        }

        return answer;
    }
}

File: 118-pascals-triangle/pascals-triangle.go
func generate(numRows int) [][]int {
    answer := make([][]int, numRows)

    for i := 0; i < numRows; i++ {
        row := make([]int, i+1)
        row[0], row[i] = 1, 1

        for j := 1; j < i; j++ {
            row[j] = answer[i-1][j-1] + answer[i-1][j]
        }

        answer[i] = row
    }

    return answer
}